src/URI.js

```diff
@@ -1223,6 +1227,26 @@
   };
 
   // compound accessors
+  p.origin = function(v, build) {
+    var parts;
+
+    if (this._parts.urn) {
+      return v === undefined ? '' : this;
+    }
+
+    if (v === undefined) {
+      return this._parts.hostname ? URI.buildOrigin(this._parts) : '';
```

Comment:
Sure, I thought it was standard to get the static function for this kind of feature. Some similar ones existed at least. Sounds fine though.